### PR TITLE
Align with `withr::local_language()` from dev withr

### LINKED ISSUE #1925: Changing language has no effect when envvar LC_ALL='C.UTF-8'
I had a lot of tests that started not passing on a Docker container. They continue to work on my local machine. When I print the traceback, it's complaining about some environment variable. 

If `lr` is a `ListReporter`, then I'm printing `lr$results$as_list()[[1]]$results`

I assume this is the result of some update. How can I cirumvent this behavior and go back to the old behavior?

```
<expectation_warning/expectation/condition>
Changing language has no effect when envvar LC_ALL='C.UTF-8'
Backtrace:
     ▆
  1. └─global calcGradesForGradescope("hw1.R", "hw1_test.R")
  2.   └─testthat::test_file(test_file, reporter = lr, env = scriptResults)
  3.     └─testthat:::test_files(...)
  4.       └─testthat:::test_files_serial(...)
  5.         ├─testthat::with_reporter(...)
  6.         │ └─base::tryCatch(...)
  7.         │   └─base (local) tryCatchList(expr, classes, parentenv, handlers)
  8.         │     └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
  9.         │       └─base (local) doTryCatch(return(expr), name, parentenv, handler)
 10.         └─base::lapply(...)
 11.           └─testthat (local) FUN(X[[i]], ...)
 12.             └─testthat::source_file(path, env = env(env), desc = desc, error_call = error_call)
 13.               └─testthat:::test_code(test = NULL, code = exprs, env = env, default_reporter = StopReporter$new())
 14.                 ├─base::tryCatch(...)
 15.                 │ └─base (local) tryCatchList(expr, classes, parentenv, handlers)
 16.                 │   ├─base (local) tryCatchOne(...)
 17.                 │   │ └─base (local) doTryCatch(return(expr), name, parentenv, handler)
 18.                 │   └─base (local) tryCatchList(expr, names[-nh], parentenv, handlers[-nh])
 19.                 │     └─base (local) tryCatchOne(expr, names, parentenv, handlers[[1L]])
 20.                 │       └─base (local) doTryCatch(return(expr), name, parentenv, handler)
 21.                 ├─base::withCallingHandlers(...)
 22.                 └─base::eval(code, test_env)
 23.                   └─base::eval(code, test_env)
 24.                     └─testthat::test_that(...) at hw1_test.R:117:1
 25.                       └─testthat:::test_code(desc, code, env = parent.frame(), default_reporter = local_interactive_reporter())
 26.                         └─testthat:::testthat_state_condition(before, after, call = sys.call(-1))
 27.                           └─testthat:::waldo_compare(before, after, x_arg = "before", y_arg = "after")
 28.                             └─testthat:::local_reporter_output()
 29.                               └─reporter$local_user_output(.env)
 30.                                 └─testthat::local_reproducible_output(...)
 31.                                   └─withr::local_language(lang, .local_envir = .env)
 32.                                     └─withr:::check_language_envvar("LC_ALL")

```

### PR DESCRIPTION
**Pull request by krlmlr**
Closes #1925

### CODE REVIEW

**Comment by krlmlr** [comment]
Confirming that this fixes my problems with dev withr: <https://github.com/r-dbi/RMariaDB/pull/305/commits/9e1f5d1631c39fef6ff6ee74f3dbe85f5ed2da6c>. I wonder why tests fail now.

**Comment by lionel-** [comment]
Maybe you need to adjust the tests like this? https://github.com/r-lib/withr/commit/08619ef7cca6ee4b01adf3d56fdebf366656ec89

**Comment by krlmlr** [comment]
Tried a moonshot, but could you please take this on (fixing tests if needed, NEWS, ...)?

File: R/local.R [hunk 1]
```diff
@@ -133,8 +133,13 @@ local_reproducible_output <- function(width = 80,
...
 
-  if (isTRUE(capabilities("NLS")) && Sys.getenv("LANG") != "C") {
-    withr::local_language(lang, .local_envir = .env)
+  if (isTRUE(capabilities("NLS"))) {
+    # FIXME: We need a new `locale` argument here, should be consistent
+    # with the `lang` argument to `local_reproducible_output()`
+    withr::local_envvar(c(LC_ALL = "en_GB"), .local_envir = .env)
+    if (!(Sys.getenv("LANG") %in% c("C", "C.UTF-8"))) {
+      withr::local_language(lang, .local_envir = .env)
+    }
```
Comment (lionel-) [inline R/local.R:142]: How about setting unconditionally? And document in `@param lang` that callers might need to set `LC_ALL` in some way, but that it's tricky to do it in a cross-platform way?

```suggestion
    suppressWarnings(
      withr::local_language(lang, .local_envir = .env)
    )
```

**Comment by hadley** [comment]
@lionel- can you take another look at this please?

**Review by lionel- — APPROVED** [review]

File: R/local.R [hunk 2]
```diff
@@ -133,8 +133,10 @@ local_reproducible_output <- function(width = 80,
     .local_envir = .env
   )
 
-  if (isTRUE(capabilities("NLS")) && Sys.getenv("LANG") != "C") {
-    withr::local_language(lang, .local_envir = .env)
+  if (isTRUE(capabilities("NLS"))) {
```
Comment (hadley) [inline R/local.R:136]: Should we also bump version in `DESCRIPTION`?

**Comment by lionel-** [comment]
Should also mention in `@param lang` that `LC_ALL` or `LANG` should be set to something other than `"C"`, which is the default in testthat environment.

I wonder if we should just deprecate that argument though. It seems impractical to use.

**Comment by lionel-** [comment]
I now think:

1. testthat should set `LANGUAGE` to `"C"` (undocumented unless I missed it but from reading gettext's source code that will work, see `guess_category_value()` in https://github.com/autotools-mirror/gettext/blob/fb38661cab30b40b503b55ac8dedcc2a8994932f/gettext-runtime/intl/dcigettext.c#L1465).

2. withr should never warn when `LANGUAGE` is set to `"C"`.

(1) is now implemented here and I've sent a patch release of withr to CRAN with (2).

**Comment by lionel-** [comment]
withr is on CRAN and I've bumped the dep before merging.

**Pull request merged**